commit aa8a87f51d7a1fb86ff75d3e3870316b6bc70dfe
Author: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date:   Tue May 31 11:51:46 2022 -0300

    misc: Use 64 bit stat for getusershell (BZ# 29204)
    
    This is a missing spot initially from 52a5fe70a2c77935.
    
    Checked on i686-linux-gnu.
    
    (cherry picked from commit ec995fb2152f160f02bf695ff83c45df4a6cd868)

diff --git a/misc/getusershell.c b/misc/getusershell.c
index 11f5aa83f888a114..4221095dca743dfa 100644
--- a/misc/getusershell.c
+++ b/misc/getusershell.c
@@ -97,7 +97,7 @@ initshells (void)
 {
 	char **sp, *cp;
 	FILE *fp;
-	struct stat64 statb;
+	struct __stat64_t64 statb;
 	size_t flen;
 
 	free(shells);
@@ -106,7 +106,7 @@ initshells (void)
 	strings = NULL;
 	if ((fp = fopen(_PATH_SHELLS, "rce")) == NULL)
 		goto init_okshells_noclose;
-	if (__fstat64(fileno(fp), &statb) == -1) {
+	if (__fstat64_time64(fileno(fp), &statb) == -1) {
 	init_okshells:
 		(void)fclose(fp);
 	init_okshells_noclose:
